src/components/document/annotation-layer.tsx

```diff
@@ -41,9 +42,24 @@ export const AnnotationLayer = observer(function AnnotationLayer({
   const ui = useUIStore();
   const persistentUI = usePersistentUIStore();
   const tileApiInterface = useContext(TileApiInterfaceContext);
+  const hotKeys = useRef(new HotKeys());
```

Comment:
One nit -- this will allocate a new `HotKeys` instance on every render (although only the first will be maintained in the ref). I would use `useMemo` here, specifically the one from `use-memo-one` so as to avoid React's annoying lifetime concerns:
```suggestion
  const hotKeys = useMemo(() => new HotKeys(), []);
```

Reply:
Makes sense.
I copied this from another tile though. We should probably fix Geometry, Draw, Graph, and Numberline in the same way.